tools/cbmc/proofs/DNS/DNSlookup/DNSlookup_harness.c

```diff
@@ -0,0 +1,18 @@
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "queue.h"
+#include "list.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_DNS.h"
+#include "FreeRTOS_IP_Private.h"
+
+// This assumes that the length of the hostname is 10 and it is null terminated.
+void harness() {
+	if( ipconfigUSE_DNS_CACHE != 0 ) {
+		char *pcHostName = malloc(10);
```

Comment:
We're going to discuss assumptions today. Does it make sense to test for the maximum permissable hostname? The [RFC for hostnames](https://tools.ietf.org/html/rfc1123#section-2) says that 255-character hostnames are allowed.

If this would totally kill performance, then perhaps you might test all hostnames up to length 10 (or whatever), plus hostnames of length 255, plus hostnames of length 256? You might use @mmuesly's configuration work to support this.